### ISSUE TITLE
Things variadic method param is "required"

### ISSUE THREAD

**Issue by robfrawley** [scrutinizer-ci/php-analyzer#763]
I have a method with the following signature:
```
public function __construct($message = null, $code = null, $previous = null, ...$replaceSet)
```

I receive the following, obviously incorrect error:

> The call to RuntimeException::__construct() misses a required argument $replaceSet.

Variadic's are implicitly not required, and thus optional parameters.

see https://scrutinizer-ci.com/g/scr-be/cache-bundle/inspections/af1ac98f-92fe-4758-a039-57ba86777f1f/issues/files/src/Component/Generator/KeyGenerator.php?status=new&selectedLabels%5B0%5D=9&orderField=path&order=asc&issueId=e0042405-6d2a-4e42-b692-e7ca5024ef86
